AndelaOSP/art-dashboard
# Feature: User can add sub category

## What does this PR do?

Enables a user with admin access to add a sub-category

## Description of Task to be completed?

**Given** A user with admin access
**When** I click on the + icon on the Asset Sub-Category column header
**And** I enter the sub-category name in a text box
**And** I select the parent category
**Then** The new sub-category should be created on the backend

## How should this be manually tested?

* Log in as an admin user
* Click on the `Assets` link in the side menu
* Click on the **+** button on the **Sub-Category** header
* Enter the required details in the modal that appears and click on **Save** button

## What are the relevant pivotal tracker stories?

[157340064](https://www.pivotaltracker.com/story/show/157340064)

## Any background context you want to add?

N/A

## Important notes

- Categories in the dropdown list are currently loaded by appending the results from the paginated end-point. This is subject to change once the end-point has been modified to fetch all the categories at once.

## Packages installed

N/A

## Deployment note

N/A

## Related PRs branch | PR (branch_name) | (pr_link)

N/A

## Todos

## Screenshots

<img width="1109" alt="screen shot 2018-07-02 at 10 22 36" src="https://user-images.githubusercontent.com/31339738/42150265-ef54e046-7de1-11e8-9ee0-9b656ed50b63.png">
